### PR TITLE
Replace request with axios

### PR DESCRIPTION
**Pull request by maxolofsson1**

### CODE REVIEW

**Review by BN-CRehn — APPROVED** [review]
🐰 🥕

File: lib/http.js [hunk 1]
```diff
@@ -3,44 +3,52 @@ const callingAppName = require(`${process.cwd()}/package.json`).name;
...
-        logger.warning("HTTP %s:%s yielded %s (error", method, url, response && response.statusCode, err, body, ")");
-        return reject(err);
-      }
+async function performRequest(method, params) {
+  const logger = buildLogger(params.correlationId, params.key);
+  const url = `${params.baseUrl || config.proxyUrl}${params.path}`;
+  logger.info(`HTTP ${method}, ${url}, params: ${JSON.stringify(params)}`);
 
-      logger.info(`HTTP response for ${method} ${url}, ${response.statusCode}, ${JSON.stringify(response.body)}`);
+  // const authHeaders = await gcpAuthHeaders(params.audience || params.baseUrl);
```
Comment (odynvolk) [inline lib/http.js:15]: Use or remove.

File: lib/http.js [hunk 2]
```diff
@@ -3,44 +3,52 @@ const callingAppName = require(`${process.cwd()}/package.json`).name;
...
+  const opts = {
+    method,
+    params: {...params.qs, ...params.query},
+    headers: buildHeaders(params, {...(params.headers || {})}),
+    validateStatus: function () {
```
Comment (odynvolk) [inline lib/http.js:20]: A little tighter. Here and further down.

```suggestion
validateStatus: () => true, // do not let axios validate status, we do that ourselves
```

File: lib/http.js [hunk 3]
```diff
@@ -70,7 +78,7 @@ function buildVerboseError(method, params, response) {
 }
 
 function dumpResponse(response) {
-  const body = (response && response.body && JSON.stringify(response.body)) || response.text;
+  const body = (response && response.data && JSON.stringify(response.data)) || response.text;
```
Comment (odynvolk) [inline lib/http.js:81]: text is not available on axios response.

```suggestion
  const body = (JSON.stringify(response?.data ?? {}));
```

File: liveness.js [hunk 4]
```diff
@@ -1,25 +1,34 @@
...
-  return new Promise((resolve, reject) => {
-    request.get(`${config.rabbit.apiUrl}/api/connections`, (err, response, body) => {
-      if (err) return reject(err);
-      if (response.statusCode !== 200) return reject(new Error(response.statusCode));
-      const myConn = JSON.parse(body).find((conn) => conn.client_properties.connection_name === config.HOSTNAME);
-      if (!myConn) return reject(new Error(`Could not find rabbit connection for: ${config.HOSTNAME}`));
-      resolve();
+async function rabbitStatus() {
+  try {
+    // console.log(`${config.rabbit.apiUrl}/api/connections`);
```
Comment (odynvolk) [inline liveness.js:8]: Remove this and line 10.

File: test/lib/liveness-test.js [hunk 5]
```diff
@@ -0,0 +1,72 @@
+"use strict";
+
+const config = require("exp-config");
+const nock = require("nock");
+const {rabbitStatus} = require("../../liveness");
+
+Feature("Liveness", () => {
```
Comment (odynvolk) [inline test/lib/liveness-test.js:7]: ⭐

**Review by odynvolk — DISMISSED** [review]
⭐ You bring gifts like cookies and this PR! Makes my day! ⭐

**Pull request merged**